src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -47,20 +47,23 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
             var isUnconstrainedTypeParameter = rewrittenLeft.Type != null && !rewrittenLeft.Type.IsReferenceType && !rewrittenLeft.Type.IsValueType;
 
             // first we can make a small optimization:
-            // If left is a constant then we already know whether it is null or not. If it is null then we 
+            // If left is a constant then we already know whether it is null or not. If it is null then we
             // can simply generate "right". If it is not null then we can simply generate
             // MakeConversion(left). This does not hold when the left is an unconstrained type parameter: at runtime,
             // it can be either left or right depending on the runtime type of T
-            if (rewrittenLeft.IsDefaultValue() && !isUnconstrainedTypeParameter)
+            if (!isUnconstrainedTypeParameter)
             {
-                return rewrittenRight;
-            }
+                if (rewrittenLeft.IsDefaultValue())
```

Comment:
Will this case get triggered if we have `(int?)(int)(default)` as the left value?

Reply:
No, actually, though this isn't affected by the PR at all.

---
In reply to: [215770773](https://github.com/dotnet/roslyn/pull/29691#discussion_r215770773) [](ancestors = 215770773)